### PR TITLE
Add mapboxApiUrl to `KeplerGL`

### PR DESCRIPTION
**Pull request by sumarlidason**
This PR adds the ability to specify the Mapbox API endpoint. Useful for folks running custom Atlas servers or on alternative country level domains such as `.cn`.

Closes https://github.com/mapbox/kepler.gl/issues/1

### CODE REVIEW

File: src/components/kepler-gl.js [hunk 1]
```diff
@@ -153,6 +154,9 @@ function KeplerGlFactory(
     }
 
     _loadMapStyle = () => {
+      if(this.props.mapStylesReplaceDefault) {
```
Comment (heshan0131) [inline src/components/kepler-gl.js]: Is the intention here to discard all default map styles provided by kepler? Not even show them in the map style selection panel? If that's the case, there should be a cleaner way to do this. See comments in `map-style-updaters.js`

File: src/reducers/map-style-updaters.js [hunk 2]
```diff
@@ -213,7 +215,8 @@ function getLayerGroupsFromStyle(style) {
 export const initMapStyleUpdater = (state, action) => ({
   ...state,
   // save mapbox access token to map style state
-  mapboxApiAccessToken: (action.payload || {}).mapboxApiAccessToken
+  mapboxApiAccessToken: (action.payload || {}).mapboxApiAccessToken,
+  mapboxApiUrl: (action.payload || {}).mapboxApiUrl
```
Comment (heshan0131) [inline src/reducers/map-style-updaters.js]: you can discard all default mapStyles here when `mapStylesReplaceDefault=true`
```js
export const initMapStyleUpdater = (state, action) => ({
  ...state,
  // save mapbox access token to map style state
  mapboxApiAccessToken: (action.payload || {}).mapboxApiAccessToken,
  mapboxApiUrl: (action.payload || {}).mapboxApiUrl,
  mapStyles: (action.payload || {}).mapStylesReplaceDefault ? {} : state.mapStyles
});
```

Comment (sumarlidason) [reply src/reducers/map-style-updaters.js]: 🍏 Done

File: src/components/kepler-gl.js (hunk 1 above)
Comment (sumarlidason) [reply src/components/kepler-gl.js]: 🍏 Done

File: test/node/reducers/map-style-test.js [hunk 3]
```diff
@@ -58,11 +58,11 @@ test('#mapStyleReducer -> INIT', t => {
     {
       ...INITIAL_MAP_STYLE,
       initialState: {},
-      mapboxApiAccessToken: 'smoothies_secret_token'
+      mapboxApiAccessToken: 'smoothies_secret_token',
+      mapboxApiUrl: undefined
```
[the remark this answers is not in the document]
Comment (sumarlidason) [reply test/node/reducers/map-style-test.js]: 🍏 Done

**Comment by sumarlidason** [comment]
@heshan0131 thank you for a quick review! 👍

**Comment by heshan0131** [comment]
@sumarlidason Unfortunately I had to edit the history of master. #575, could you please update your local forked master and rebase? Sorry for the inconvenience.

**Comment by heshan0131** [comment]
Thanks for updating the branch. One more thing, to signoff your commit:

```
git commit --amend --signoff

git push --force-with-lease origin feature_1b
```

This will get DCO to pass. Thanks!

**Pull request merged**